frontend/test/pytest/test_gradient.py

```diff
@@ -1230,5 +1230,24 @@ def cir(x: float):
                 return grad(g)(x)
 
 
+class TestGradientUsagePatterns:
+    """Test usage patterns of Gradient functions"""
+
+    def test_grad_usage_patterns(self):
+        """Test usage patterns of catalyst.grad."""
+
+        def fn(x):
+            return x**2
+
+        x = 4.0
+
+        res_pattern_fn_as_argument = grad(fn)(x)
+        res_pattern_partial = grad()(fn)(x)
+        expected = jax.grad(fn)(x)
```

Comment:
Mmm... I know this is in the ticket, but if feels weird that `grad` does not take any values whatsoever. I don't like this and I will defer to @dime10 to decide on whether this is a good UI.

Reply:
It's to enable this pattern:
```py
@grad(method="fd")
def f(x):
  ...
  
f(x)
```

I think Raul is just omitting the decorator syntax (that is `grad(method="fd")(fn)(x)` is equivalent to what I wrote), and also left the keyword parameters at their default value, but otherwise they are testing the same thing. 

Admittedly using some keyword arguments would make the test clearer and would also test that capturing the kwargs via `locals` is working properly.